geo/src/main/java/org/odk/collect/geo/selection/SelectionMapFragment.kt

```diff
@@ -209,14 +205,37 @@ class SelectionMapFragment(
         )
     }
 
-    private fun restoreZoomFromPreviousState(state: Bundle) {
+    private fun initializeViewport() {
+        previousState?.let {
+            if (restoreZoomFromPreviousState(it)) {
+                viewportInitialized = true
+            }
+        }
+
+        if (!viewportInitialized) {
+            itemsByFeatureId.filter { it.value.selected }.map { it.key }.firstOrNull()?.let {
+                onFeatureClicked(it)
+                viewportInitialized = true
+            }
+        }
+
+        if (!viewportInitialized && zoomToFeatureBoundingBoxOnLoad && points.isNotEmpty()) {
+            map.zoomToBoundingBox(points, 0.8, false)
+            viewportInitialized = true
+        }
+
+        map.setGpsLocationListener { point -> onLocationChanged(point) }
```

Comment:
`onLocationChanged` could set the gps location listener to null once the viewport is initialized, I think.

As I mentioned in https://github.com/getodk/collect/pull/5098#issuecomment-1097281775, maybe this whole strategy is problematic. An alternative would be to see if `map.getGpsLocation` is set and if so, center to that, otherwise fall back to the bounding box and forget about having a listener at all.